docs/reference/migration/apis/data-stream-reindex.asciidoc

```diff
@@ -0,0 +1,358 @@
+
+Later, once all the backing indices have finished, the <<data-stream-reindex-status-api,reindex status API>> will return something like the following:
+
+[source,console-result]
+----
+{
+  "start_time_millis": 1737676174349,
+  "complete": true,
+  "total_indices_in_data_stream": 4,
+  "total_indices_requiring_upgrade": 2,
+  "successes": 2,
+  "in_progress": [],
+  "pending": 0,
+  "errors": []
+}
+----
+// TEST[skip:specific value is part of explanation]
+
+Notice that the value of `total_indices_requiring_upgrade` is `2`, unlike the previous status, which had a value of `3`.
+This is because `.ds-my-data-stream-2025.01.23-000001` was upgraded before the task cancellation.
+After the restart, the API sees that it does not need to be upgraded, thus does not include it in `total_indices_requiring_upgrade` or `successes`,
+despite the fact that it upgraded successfully.
+
+The completed status will be accessible from the status API for 24 hours after completion of the task.
+
+We can now check the data stream to verify that indices were upgraded:
+
+[source,console]
+----
+GET _data_stream/my-data-stream?filter_path=data_streams.indices.index_name
+----
+// TEST[continued]
+
+
+which returns:
+[source,console-result]
+----
+{
+  "data_streams": [
+    {
+      "indices": [
+        {
+          "index_name": ".migrated-ds-my-data-stream-2025.01.23-000003"
+        },
+        {
+          "index_name": ".migrated-ds-my-data-stream-2025.01.23-000002"
+        },
+        {
+          "index_name": ".migrated-ds-my-data-stream-2025.01.23-000001"
+        },
+        {
+          "index_name": ".ds-my-data-stream-2025.01.23-000004"
+        }
+      ]
+    }
+  ]
+}
+----
+// TEST[skip:did not actually run reindex]
+
+Index `.ds-my-data-stream-2025.01.23-000004` is the write index and didn't need to be upgraded because it was created with version 8.x.
+The other three backing indices are now prefixed with `.migrated` because they have been upgraded.
+
+We can now check the indices and verify that they have version 8.x:
+[source,console]
+----
+GET .migrated-ds-my-data-stream-2025.01.23-000001?human&filter_path=*.settings.index.version.created_string
+----
+// TEST[skip:migrated index does not exist]
+
+which returns:
+[source,console-result]
+----
+{
+  ".migrated-ds-my-data-stream-2025.01.23-000001": {
+    "settings": {
+      "index": {
+        "version": {
+          "created_string": "8.18.0"
+        }
+      }
+    }
+  }
+}
+----
+// TEST[skip:migrated index does not exist]
+
+[[reindex-data-stream-handling-failure]]
+===== Handling Failures
+Since the reindex data stream API runs in the background, failure information can be obtained through the <<data-stream-reindex-status-api,reindex status API>>.
+For example, if the backing index `.ds-my-data-stream-2025.01.23-000002` was accidentally deleted, we would see a status like the following:
```

Comment:
`if the backing index `.ds-my-data-stream-2025.01.23-000002` was accidentally deleted` -- maybe add `by a user` or something like that to avoid making it sound like this process might accidentally delete data.